src/libraries/System.Text.RegularExpressions/tests/FunctionalTests/RegexGeneratorOutputTests.cs

```diff
@@ -0,0 +1,875 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Text.RegularExpressions.Tests
+{
+    [ConditionalClass(typeof(PlatformDetection), nameof(PlatformDetection.IsReflectionEmitSupported), nameof(PlatformDetection.IsNotMobile), nameof(PlatformDetection.IsNotBrowser))]
+    public partial class RegexGeneratorOutputTests
+    {
+        // This exists to ensure we're aware of any egregious breaks to formatting / readability.
+        // Any updates that impact the generated code in these baselines will need to be updated
+        // as changes are made to the code emitted by the generator.
+
+        [Theory]
+        [MemberData(nameof(ValidateExpectedOutput_MemberData))]
+        public async Task ValidateExpectedOutput(string program, string expected)
+        {
+            string actual = await RegexGeneratorHelper.GenerateSourceText(program, allowUnsafe: true, checkOverflow: false);
+
+            Console.OutputEncoding = Encoding.UTF8;
+
+            expected = NormalizeWhiteSpace(expected);
+            actual = NormalizeWhiteSpace(actual);
+            AssertExtensions.Equal(expected, actual);
+
+            static string NormalizeWhiteSpace(string code)
+            {
+                var input = new StringReader(code);
+                var output = new StringBuilder();
+
+                string line;
+                while ((line = input.ReadLine()) != null)
+                {
+                    if (string.IsNullOrWhiteSpace(line))
+                    {
+                        line = "";
+                    }
+
+                    output.AppendLine(line);
+                }
+
+                return output.ToString();
+            }
+        }
+
+        public static IEnumerable<object[]> ValidateExpectedOutput_MemberData()
+        {
+            yield return new object[]
+            {
+                """
+                using System.Text.RegularExpressions;
+                partial class C
+                {
+                    [GeneratedRegex(@"^(?<proto>\w+)://[^/]+?(?<port>:\d+)?/")]
+                    public static partial Regex Valid();
+                }
+                """,
+
+                """
+                // <auto-generated/>
```

Comment:
For the Logging Generator we put the baselines in separate files. See https://github.com/dotnet/runtime/tree/main/src/libraries/Microsoft.Extensions.Logging.Abstractions/tests/Microsoft.Extensions.Logging.Generators.Tests/Baselines.

And then have a "compare baselines" function that tells you the diff

https://github.com/dotnet/runtime/blob/31e4f404c218eae7ba999c4df5346d30f971451c/src/libraries/Microsoft.Extensions.Logging.Abstractions/tests/Microsoft.Extensions.Logging.Generators.Tests/LoggerMessageGeneratorEmitterTests.cs#L166-L206

What do you think about doing it similarly here?

Reply:
You may need something like this because of the `42.42.42.42` version in the generated output.

Reply:
Seems like 6 of one, half dozen of another, especially with raw string literals as are being used here.  What do you see as the advantage of having the separate files?

Reply:
1. When some thing fails, the above comparison code will tell you which line it failed on, and a diff of the before/after on just that line.
2. If we wanted to get fancier in the future, one approach I like to take with baseline files is to write the current output to a file, and then use a comparison tool (like Beyond Compare) to diff the full file, or just `cp current.generated.txt baseline.generated.txt` to easily update the baseline.
3. It keeps the code cleaner - since the generated output is in a separate file.

Reply:
One of the advantages of having both "gold" and "temp output" files is that there is no visual noise. Only what is expected. This greatly simplifies diffing, updating baselines from temp to gold, etc. This kind of thing is called snapshot based testing as I recall.

Reply:
> When some thing fails, the above comparison code will tell you which line it failed on, and a diff of the before/after on just that line.

Isn't that just an artifact of the CompareLines method, which is operating over the data in memory?  It doesn't matter or know where that data originally came from.

> If we wanted to get fancier in the future, one approach I like to take with baseline files is to write the current output to a file, and then use a comparison tool (like Beyond Compare) to diff the full file, or just cp current.generated.txt baseline.generated.txt to easily update the baseline.

If we did that, we could just as easily write both strings out for comparison.

> You may need something like this because of the 42.42.42.42 version in the generated output.

In what situation will our runtime tests use a different version?

> It keeps the code cleaner - since the generated output is in a separate file.

I personally see it the other way.  Having to go look at a bunch of other files to see what the input and output are feels very clunky to me.

Reply:
> In what situation will our runtime tests use a different version?

See https://github.com/dotnet/runtime/issues/58175 and https://github.com/dotnet/runtime/pull/58220 for when it broke in the Logging generator.

> If we did that, we could just as easily write both strings out for comparison.

But how can you easily update the baseline? You need to surgically do it inside the test code file, as opposed to just a simple file copy (or two if you changed both tests).

> It doesn't matter or know where that data originally came from.

It doesn't matter where it came from. But a big point is we can just reuse what's already there for the Logging generator (or enhance it and make them both better).

Reply:
> In what situation will our runtime tests use a different version?

Dev builds always use 42.42.42.42, but some ci builds, rolling builds, and official builds would not.

Reply:
> But how can you easily update the baseline?

The exact same way, copy paste. These are raw string literals, the contents is exactly what it is, exactly what would be in a different file.

> But a big point is we can just reuse what's already there for the Logging generator (or enhance it and make them both better).

Which is fine feedback but has nothing to do with whether the code be in one file or separate files. Am I misunderstanding?

> Dev builds always use 42.42.42.42, but some ci builds, rolling builds, and official builds would not.

Ok, I'll fix the version.

Reply:
It may just be personal opinion, but from my experience I think there are advantages to splitting the baseline contents into a separate file, especially when diffing and updating the baselined contents. I've not convinced you of this, but I'm not going to block on it.

Reply:
> from my experience I think there are advantages to splitting the baseline contents into a separate file, especially when diffing and updating the baselined contents

I agree with that pre- raw string literals, where you'd need to do surgery to update various tokens every time you copy-and-paste the new baseline.  raw string literals change the game, IMO, as there's no surgery needed.

Reply:
With the current code, when you copy-paste the generated output, will you still need to `tab` the string contents over from the generated output? It looks like the strings start 16 spaces from the left side of the file.

Reply:
>  when you copy-paste the generated output, will you still need to tab the string contents over from the generated output?

No, VS does that automatically.